### PR TITLE
fix(): catch unhandled exceptions

### PR DESCRIPTION
**Pull request by pooreumu**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

not catch unhandled exceptions

Issue Number: N/A

## What is the new behavior?
catch unhandled exceptions

## Does this PR introduce a breaking change?
- [x] Yes
- [ ] No

### CODE REVIEW

File: tests/e2e/module-e2e.spec.ts [hunk 1]
```diff
@@ -122,6 +122,17 @@ describe('EventEmitterModule - e2e', () => {
     expect(customConsumer.isEmitted).toBeTruthy();
   });
 
+  it('should be able to handled when an unexpected error occurs', async () => {
```
Comment (kamilmysliwiec) [inline tests/e2e/module-e2e.spec.ts:125]:
```suggestion
  it('should be able to gracefully recover when an unexpected error occurs', async () => {
```

Comment (pooreumu) [reply tests/e2e/module-e2e.spec.ts]: Thanks for the suggestion.

File: lib/decorators/on-event.decorator.ts [hunk 2]
```diff
@@ -21,6 +22,22 @@ export interface OnEventMetadata {
  */
 export type OnEventType = string | symbol | Array<string | symbol>;
 
+/**
+ * Wraps the method in try/catch blocks.
+ * @param descriptor
+ */
+const wrapDescriptorInTryCatchBlocks = (descriptor: any) => {
+  const originMethod = descriptor.value;
```
Comment (kamilmysliwiec) [inline lib/decorators/on-event.decorator.ts:30]: overriding the descriptor will consequently make other metadata (attached through `@SetMetadata`) lost for this method

Comment (pooreumu) [reply lib/decorators/on-event.decorator.ts:30]: I expect to be able to maintain through line 38(`Object.setPrototypeOf(descriptor.value, originMethod);`).

Comment (kamilmysliwiec) [reply lib/decorators/on-event.decorator.ts:30]: Wouldn't it be easier to just add try-catch here https://github.com/nestjs/event-emitter/blob/master/lib/event-subscribers.loader.ts#L95 and here https://github.com/nestjs/event-emitter/blob/c25acc940103738f445decf1b14ee23347845ba5/lib/event-subscribers.loader.ts#L138

Comment (pooreumu) [reply lib/decorators/on-event.decorator.ts]: Thanks for the suggestion.

**Comment by kamilmysliwiec** [comment]
LGTM

**Pull request merged**